python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -7,14 +7,38 @@
 
 
 class BatchMapper(Preprocessor):
-    """Apply ``fn`` to batches of records of given dataset.
-
-    This is meant to be generic and supports low level operation on records.
-    One could easily leverage this preprocessor to achieve operations like
-    adding a new column or modifying a column in place.
+    """Apply an arbitrary operation to a dataset.
+
+    :class:`BatchMapper` applies a user-defined function to batches of a dataset. A
+    batch is a Pandas ``DataFrame`` that represents a small amount of data. By modifying
+    batches instead of individual records, this class can efficiently transform a
+    dataset with vectorized operations.
+
+    Use this preprocessor to apply stateless operations that aren't already built-in.
+
+    .. tip::
+        :class:`BatchMapper` doesn't need to be fit. You can call
+        ``transform`` without calling ``fit``.
+
+    Examples:
+        Use :class:`BatchMapper` to apply arbitrary operations like dropping a column.
+
+        >>> import pandas as pd
+        >>> import ray
+        >>> from ray.data.preprocessors import BatchMapper
+        >>>
+        >>> df = pd.DataFrame({"X": [0, 1, 2], "Y": [3, 4, 5]})
+        >>> ds = ray.data.from_pandas(df)  # doctest: +SKIP
+        >>>
+        >>> def fn(batch: pd.DataFrame) -> pd.DataFrame:
+        ...     return batch.drop("Y", axis="columns")
```

Comment:
nit; ideally we choose an example that is actually not supported by Ray

Reply:
How do you drop a column without `BatchMapper`?